File: clients/fireback-ios/firebackios/sdk/modules/iot/ModbusTaskEntity.swift
import Foundation
class ModbusTaskEntity : Codable, Identifiable {
    var name: String? = nil
    var modbusId: Int? = nil
    var device: DeviceEntity? = nil
    // var deviceId: String? = nil
    var connectionType: ModbusConnectionTypeEntity? = nil
    // var connectionTypeId: String? = nil
    var functionCode: ModbusFunctionCodeEntity? = nil
    // var functionCodeId: String? = nil
    var register: Int? = nil
    var writeInterval: Int? = nil
    var readInterval: Int? = nil
    var range: Int? = nil
    var length: Int? = nil
    var variableType: ModbusVariableTypeEntity? = nil
    // var variableTypeId: String? = nil
}
class ModbusTaskEntityViewModel: ObservableObject {
  // improve the fields here
  @Published var name: String? = nil
  @Published var nameErrorMessage: String? = nil
  @Published var modbusId: Int? = nil
  @Published var modbusIdErrorMessage: Int? = nil
  @Published var register: Int? = nil
  @Published var registerErrorMessage: Int? = nil
  @Published var writeInterval: Int? = nil
  @Published var writeIntervalErrorMessage: Int? = nil
  @Published var readInterval: Int? = nil
  @Published var readIntervalErrorMessage: Int? = nil
  @Published var range: Int? = nil
  @Published var rangeErrorMessage: Int? = nil
  @Published var length: Int? = nil
  @Published var lengthErrorMessage: Int? = nil
  func getDto() -> ModbusTaskEntity {
      var dto = ModbusTaskEntity()
    dto.name = self.name
    dto.modbusId = self.modbusId
    dto.register = self.register
    dto.writeInterval = self.writeInterval
    dto.readInterval = self.readInterval
    dto.range = self.range
    dto.length = self.length
      return dto
  }
}

File: clients/android-fireback/app/src/main/java/com/fireback/modules/iot/ModbusTaskEntity.java
package com.fireback.modules.iot;
import com.fireback.modules.workspaces.OkayResponseDto;
import com.google.gson.Gson;
import com.fireback.JsonSerializable;
import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.ViewModel;
public class ModbusTaskEntity extends JsonSerializable {
    public String name;
    public int modbusId;
    public DeviceEntity device;
    public ModbusConnectionTypeEntity connectionType;
    public ModbusFunctionCodeEntity functionCode;
    public int register;
    public int writeInterval;
    public int readInterval;
    public int range;
    public int length;
    public ModbusVariableTypeEntity variableType;
    public static class VM extends ViewModel {
    // upper: Name name
    private MutableLiveData< String > name = new MutableLiveData<>();
    public MutableLiveData< String > getName() {
        return name;
    }
    public void setName( String  v) {
        name.setValue(v);
    }
    // upper: ModbusId modbusId
    private MutableLiveData< Integer > modbusId = new MutableLiveData<>();
    public MutableLiveData< Integer > getModbusId() {
        return modbusId;
    }
    public void setModbusId( Integer  v) {
        modbusId.setValue(v);
    }
    // upper: Device device
    private MutableLiveData< DeviceEntity > device = new MutableLiveData<>();
    public MutableLiveData< DeviceEntity > getDevice() {
        return device;
    }
    public void setDevice( DeviceEntity  v) {
        device.setValue(v);
    }
    // upper: ConnectionType connectionType
    private MutableLiveData< ModbusConnectionTypeEntity > connectionType = new MutableLiveData<>();
    public MutableLiveData< ModbusConnectionTypeEntity > getConnectionType() {
        return connectionType;
    }
    public void setConnectionType( ModbusConnectionTypeEntity  v) {
        connectionType.setValue(v);
    }
    // upper: FunctionCode functionCode
    private MutableLiveData< ModbusFunctionCodeEntity > functionCode = new MutableLiveData<>();
    public MutableLiveData< ModbusFunctionCodeEntity > getFunctionCode() {
        return functionCode;
    }
    public void setFunctionCode( ModbusFunctionCodeEntity  v) {
        functionCode.setValue(v);
    }
    // upper: Register register
    private MutableLiveData< Integer > register = new MutableLiveData<>();
    public MutableLiveData< Integer > getRegister() {
        return register;
    }
    public void setRegister( Integer  v) {
        register.setValue(v);
    }
    // upper: WriteInterval writeInterval
    private MutableLiveData< Integer > writeInterval = new MutableLiveData<>();
    public MutableLiveData< Integer > getWriteInterval() {
        return writeInterval;
    }
    public void setWriteInterval( Integer  v) {
        writeInterval.setValue(v);
    }
    // upper: ReadInterval readInterval
    private MutableLiveData< Integer > readInterval = new MutableLiveData<>();
    public MutableLiveData< Integer > getReadInterval() {
        return readInterval;
    }
    public void setReadInterval( Integer  v) {
        readInterval.setValue(v);
    }
    // upper: Range range
    private MutableLiveData< Integer > range = new MutableLiveData<>();
    public MutableLiveData< Integer > getRange() {
        return range;
    }
    public void setRange( Integer  v) {
        range.setValue(v);
    }
    // upper: Length length
    private MutableLiveData< Integer > length = new MutableLiveData<>();
    public MutableLiveData< Integer > getLength() {
        return length;
    }
    public void setLength( Integer  v) {
        length.setValue(v);
    }
    // upper: VariableType variableType
    private MutableLiveData< ModbusVariableTypeEntity > variableType = new MutableLiveData<>();
    public MutableLiveData< ModbusVariableTypeEntity > getVariableType() {
        return variableType;
    }
    public void setVariableType( ModbusVariableTypeEntity  v) {
        variableType.setValue(v);
    }
    }
}